### PR TITLE
Isolate user hadoop dependencies from cluster environment

### PR DESCRIPTION
**Pull request by plliao**
Isolate user hadoop dependencies from cluster environment

For more details, Please see #450

### CODE REVIEW

File: tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java [hunk 1]
```diff
@@ -134,6 +134,10 @@ private TonyConfigurationKeys() {
   public static final String APPLICATION_DISTRIBUTED_MODE = TONY_APPLICATION_PREFIX + "distributed-mode";
   public static final String DEFAULT_APPLICATION_DISTRIBUTED_MODE = DistributedMode.GANG.name();
 
+  public static final String TONY_APPLICATION_MAPREDUCE_PREFIX = TONY_APPLICATION_PREFIX + "mapreduce.";
+  public static final String APPLICATION_MAPREDUCE_FRAMEWORK_PATH = TONY_APPLICATION_MAPREDUCE_PREFIX + "framework.path";
```
Comment (hungj) [inline tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java:138]: i think we can scrap TONY_APPLICATION_MAPREDUCE_PREFIX and just make this TONY_APPLICATION_PREFIX + framework.path - we can rename the variable APPLICATION_FRAMEWORK_PATH too

File: tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java [hunk 2]
```diff
@@ -134,6 +134,10 @@ private TonyConfigurationKeys() {
...
+  public static final String APPLICATION_MAPREDUCE_CLASSPATH = TONY_APPLICATION_MAPREDUCE_PREFIX + "classpath";
```
Comment (hungj) [inline tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java:139]: same here, also rename APPLICATION_FRAMEWORK_CLASSPATH?

File: tony-core/src/main/java/com/linkedin/tony/TonyClient.java [hunk 3]
```diff
@@ -138,6 +142,21 @@
   // For access from CLI.
   private Set<TaskInfo> taskInfos = new HashSet<>();
 
+  /**
+   * Gets default MapReduce Framework classpath derived from yarnConf.
+   */
+  public static String getDefaultMapReduceFrameworkClasspath(Configuration yarnConf) {
```
Comment (hungj) [inline tony-core/src/main/java/com/linkedin/tony/TonyClient.java:148]: getDefaultYarnClasspath?

File: tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java (hunk 1 above)
Comment (plliao) [reply tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java:138]: Yeah, I thought about it. But we have another configuration, "tony.application.framework" used to decide the deep learning framework, e.g., TensorFlow or PyTorch. I am wondering if they are too similar. (maybe not?) Will there be a better name?

Comment (hungj) [reply tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java]: Makes sense. How about `tony.application.dependencies.location` and `tony.application.dependencies.classpath`? I'd prefer not to have mapreduce in the name, logically they should be separate

Comment (plliao) [reply tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java]: I update the dependencies with hadoop. Please let me know if it works for you. Thanks!

File: tony-core/src/main/java/com/linkedin/tony/TonyClient.java (hunk 3 above)
Comment (plliao) [reply tony-core/src/main/java/com/linkedin/tony/TonyClient.java]: Updated

File: tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java (hunk 2 above)
Comment (plliao) [reply tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java]: Updated.

**Review by hungj — APPROVED** [review]

**Comment by oliverhu** [comment]
Great work @plliao !! Congrats on the first PR merged!!

**Pull request merged**